frontend/src/metabase/visualizations/components/settings/ChartSettingFieldsPicker.jsx

```diff
@@ -10,62 +11,106 @@ const ChartSettingFieldsPicker = ({
   onChange,
   addAnother,
   ...props
-}) => (
-  <div>
-    {Array.isArray(value) ? (
-      value.map((v, index) => (
-        <ChartSettingFieldPicker
-          {...props}
-          className={index > 0 ? "mt1" : null}
-          key={index}
-          value={v}
-          options={options}
-          onChange={v => {
-            const newValue = [...value];
-            // this swaps the position of the existing value
-            const existingIndex = value.indexOf(v);
-            if (existingIndex >= 0) {
-              newValue.splice(existingIndex, 1, value[index]);
-            }
-            // replace with the new value
-            newValue.splice(index, 1, v);
-            onChange(newValue);
-          }}
-          onRemove={
-            value.filter(v => v != null).length > 1 ||
-            (value.length > 1 && v == null)
-              ? () =>
-                  onChange([
-                    ...value.slice(0, index),
-                    ...value.slice(index + 1),
-                  ])
-              : null
-          }
-        />
-      ))
-    ) : (
-      <span className="text-error">{t`error`}</span>
-    )}
-    {addAnother && (
-      <AddAnotherContainer>
-        <a
-          className="text-brand text-bold py1"
-          onClick={() => {
-            const remaining = options.filter(o => value.indexOf(o.value) < 0);
-            if (remaining.length === 1) {
-              // if there's only one unused option, use it
-              onChange(value.concat([remaining[0].value]));
-            } else {
-              // otherwise leave it blank
-              onChange(value.concat([undefined]));
-            }
-          }}
-        >
-          {addAnother}
-        </a>
-      </AddAnotherContainer>
-    )}
-  </div>
-);
+}) => {
+  const handleDragEnd = ({ source, destination }) => {
+    const oldIndex = source.index,
+      newIndex = destination.index;
+
+    const valueCopy = [...value];
+    valueCopy.splice(newIndex, 0, valueCopy.splice(oldIndex, 1)[0]);
+    onChange(valueCopy);
+  };
+
+  const calculateOptions = item => {
```

Comment:
nit: this introduces a new variable named `item` which has the same type as `v`. This kind of indicates that `v` isn't a great name here, as `item` and `v` in this function should be named the same to communicate they're of the same type.

I've tried refactoring this function using the suggested names from other comments please let me know if this is easier to understand.

```ts
  const calculateOptions = field => {
    return options.filter(
      option =>
        fields.findIndex(field => field === option.value) < 0 ||
        option.value === field,
    );
  };
```